x/params/msgstat/ante.go

```diff
@@ -0,0 +1,25 @@
+package msgstat
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	params "github.com/cosmos/cosmos-sdk/x/params/store"
+)
+
+// NewAnteHandler returns an AnteHandler that checks
+// whether msg type is activate or not
+func NewAnteHandler(store params.Store) sdk.AnteHandler {
+	return func(ctx sdk.Context, tx sdk.Tx) (sdk.Context, sdk.Result, bool) {
+		for _, msg := range tx.GetMsgs() {
+			key := params.NewKey(msg.Type())
+			if !store.Has(ctx, key) {
+				return ctx, sdk.ErrUnauthorized("deactivated msg type").Result(), true
+			}
+			var activated bool
+			store.Get(ctx, key, &activated)
```

Comment:
It seems like the following would might be cleaner:

`activated, ok := store.GetBool(ctx, key)`

Then we wouldn't need a separate Has call.

Reply:
Added GetIfExists - it does not modify pointer if the value is nil